### ISSUE TITLE
Panic when attempting to view metrics for workload.

### ISSUE THREAD

**Issue by DominicGunn** [rancher/rancher#19358]
**What kind of request is this (question/bug/enhancement/feature request):**

This is a bug.

**Steps to reproduce (least amount of steps as possible):**

Running Rancher against a cluster that has 40+ nodes, 400+ workloads, then attempting to view metrics for the workload, rancher Panics.

**Result:**

```
[rancher-79c6954c7c-wwgg9] 2019/04/03 11:27:38 [ERROR] Panic serving api request:
[rancher-79c6954c7c-wwgg9] goroutine 11566 [running]:
[rancher-79c6954c7c-wwgg9] runtime/debug.Stack(0xc00c7e9a70, 0x33da1e0, 0x6a45ef0)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x419c280, 0xc00bb92a80)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:175 +0x67
[rancher-79c6954c7c-wwgg9] panic(0x33da1e0, 0x6a45ef0)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/runtime/panic.go:513 +0x1b9
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/api/customization/monitor.parseMetricParams(0xc004494b00, 0x416bc80, 0xc007348648, 0xc00ce02958, 0x8, 0xc006b478f0, 0x7, 0xc00ad71a40, 0xc00ba18c60, 0x12, ...)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/handler_utils.go:253 +0x1594
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/api/customization/monitor.graph2Metrics(0xc004494b00, 0x4212840, 0xc001488600, 0xc006b478f0, 0x7, 0xc00ce02958, 0x8, 0xc008aba2e0, 0x1a, 0xc00b892660, ...)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/cluster_graph_action.go:146 +0xf3
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/api/customization/monitor.(*ProjectGraphHandler).QuerySeriesAction(0xc0006a2200, 0xc00903eba5, 0x5, 0xc00a645f60, 0xc00ad586c0, 0x0, 0x0)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/project_graph_action.go:89 +0x79b
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/api/customization/monitor.(*ProjectGraphHandler).QuerySeriesAction-fm(0xc00903eba5, 0x5, 0xc00a645f60, 0xc00ad586c0, 0x0, 0x0)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/api/server/managementstored/setup.go:511 +0x52
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/rancher/norman/api.handleAction(0xc00a645f60, 0xc00ad586c0, 0x0, 0x0)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:263 +0x62
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).handle(0xc000b84420, 0x419c280, 0xc00bb92a80, 0xc00d848e00, 0xc00c68d080, 0xc001515050, 0xc00eeb5738)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:251 +0xd2
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP(0xc000b84420, 0x419c280, 0xc00bb92a80, 0xc00d848e00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:180 +0x7d
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0006ca8c0, 0x419c280, 0xc00bb92a80, 0xc00d848e00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:159 +0xf1
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/auth/requests.authHeaderHandler.ServeHTTP(0x41666c0, 0xc0015e5920, 0x415bec0, 0xc0006ca8c0, 0x417a340, 0xc000ec2a00, 0x4166680, 0xc001515710, 0x419c280, 0xc00bb92a80, ...)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/auth/requests/filter.go:110 +0x60d
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/audit.auditHandler.ServeHTTP(0x415a800, 0xc001193640, 0x0, 0x419c280, 0xc00bb92a80, 0xc00d848a00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/audit/filter.go:31 +0x46a
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0006ca850, 0x419c280, 0xc00bb92a80, 0xc00d848a00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:159 +0xf1
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/dynamiclistener.httpRedirect.func1(0x419c280, 0xc00bb92a80, 0xc004d18c00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/dynamiclistener/server.go:489 +0x278
[rancher-79c6954c7c-wwgg9] net/http.HandlerFunc.ServeHTTP(0xc001100d60, 0x419c280, 0xc00bb92a80, 0xc004d18c00)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/net/http/server.go:1964 +0x44
[rancher-79c6954c7c-wwgg9] github.com/rancher/rancher/pkg/dynamiclistener.(*Server).cacheIPHandler.func1(0x419c280, 0xc00bb92a80, 0xc004d18c00)
[rancher-79c6954c7c-wwgg9] 	/go/src/github.com/rancher/rancher/pkg/dynamiclistener/server.go:419 +0x101
[rancher-79c6954c7c-wwgg9] net/http.HandlerFunc.ServeHTTP(0xc001100d80, 0x419c280, 0xc00bb92a80, 0xc004d18c00)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/net/http/server.go:1964 +0x44
[rancher-79c6954c7c-wwgg9] net/http.serverHandler.ServeHTTP(0xc0025b6b60, 0x419c280, 0xc00bb92a80, 0xc004d18c00)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/net/http/server.go:2741 +0xab
[rancher-79c6954c7c-wwgg9] net/http.(*conn).serve(0xc0012fde00, 0x41a5900, 0xc011054a40)
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/net/http/server.go:1847 +0x646
[rancher-79c6954c7c-wwgg9] created by net/http.(*Server).Serve
[rancher-79c6954c7c-wwgg9] 	/usr/local/go/src/net/http/server.go:2851 +0x2f5
[rancher-79c6954c7c-wwgg9]
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.1, also on v2.2.0.
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Imported.
- Machine type (cloud/VM/metal) and specifications (CPU/memory): AWS
- Kubernetes version (use `kubectl version`): v1.13.4

**Comment by loganhz** [comment]
I can reproduce this issue if I import the following deployment

```
controllers/nginx-deployment.yaml 

apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80

```

**Comment by jiaqiluo** [comment]
The bug fix is validated with Rancher: master  4fd25480
Setup: single install; an EC2 cluster with 4 nodes( 1 etcd, 1 control plane, 2 workers)
Steps are the same as the comment here:https://github.com/rancher/rancher/issues/19371#issuecomment-479810945
